step/serialization.md

```diff
@@ -260,31 +270,31 @@ public class ClassSerializationClient {
 </copy>
 ```
 
-Now run `ClassSerializationClient`:
+Now run `SerializationClient`:
 
 ```
 <copy>
-java ClassSerializationClient.java
+java SerializationClient.java
 </copy>
 ```
 
-Which will throw a NullPointerException, when in `ClassSerializationServer` `toUpperCase()` is called: 
+This will cause `SerializationServer` to throw a `NullPointerException` and a very confused developer as it should be possible for the field `message` to be null! 
 
 ```no-highlight
 Exception in thread "main" java.lang.NullPointerException: Cannot invoke "String.toUpperCase()" because the return value of "Object.toString()" is null
-	at ClassSerializationServer.main(ClassSerializationServer.java:32)
+	at SerializationServer.main(SerializationServer.java:32)
 ``` 
 
-This happens because in Java when a stream is being deserialized, the constructor for the class is not called and the fields are set reflectively. So even though in the version of `MessageClass` being used by `ClassSerializationServer` it should programmatically be impossible for `message` to be null. Serialization is able to break encapsulation and provide a null value, creating a difficult to track down bug. 
+This happens because in Java when a stream is being deserialized, the constructor for the class is not called and the fields are set reflectively. So even though in the version of `Message` being used by `SerializationServer` it should programmatically be impossible for `message` to be null. Serialization is able to break encapsulation and set the `message` field to null, creating a difficult to track down bug. 
```

Comment:
the constructor for the class is not called and the fields are set reflectively.
-> the constructor for the class is not called as the fields are set reflectively.